cupy/_core/_routines_binary.pyx

```diff
@@ -64,6 +64,37 @@ cdef _invert = create_ufunc(
     ''')
 
 
+cdef _bitwise_count = create_ufunc(
+    'cupy_bitwise_count',
+    ('b->B', 'B->B', 'h->B', 'H->B', 'i->B', 'I->B', 'l->B', 'L->B', 'q->B', 'Q->B'),
+    'out0 = _cupy_bitcount(in0);',
+    doc='''Computes the number of 1-bits in each element.
+
+    Only integer arrays are handled.
+
+    .. seealso:: :data:`numpy.bit_count`
+
+    ''',
+    preamble='''
+template <typename T>
+__device__ inline unsigned int _cupy_bitcount(T x) {
```

Comment:
It seems this incurs extra type casts which is unnecessary?
- popc returns `int` (or int32)
- This cast int32 to uint32
- The ufunc casts uint32 to uint8 (as required by NumPy)

```suggestion
__device__ inline int _cupy_bitcount(T x) {
```